cranelift/filetests/src/test_compile.rs

```diff
@@ -63,15 +67,25 @@ impl SubTest for TestCompile {
         info!("Generated {} bytes of code:\n{}", total_size, disasm);
 
         if self.precise_output {
-            check_precise_output(&disasm, context)
+            check_precise_output(isa, params, &compiled_code, context)
         } else {
             run_filecheck(&disasm, context)
         }
     }
 }
 
-fn check_precise_output(text: &str, context: &Context) -> Result<()> {
-    let actual = text.lines().collect::<Vec<_>>();
+fn check_precise_output(
+    isa: &dyn isa::TargetIsa,
+    params: FunctionParameters,
```

Comment:
It looks like the only use of `params` in this function is to borrow it, so you could make this argument a borrow as well. Then in `run` I think you can pass `&comp_ctx.func.params` instead of having to clone the params.

Reply:
Nice!

Reply:
Oh, there's a borrow checker error here if I don't clone :(

Reply:
Huh, I may have misread something. Did you access the `func` through `comp_ctx`, where it's been moved?

Reply:
`compile` takes a `mut self`, so `comp_ctx` is no longer available at that point.

Reply:
I misread my earlier error, it's not taking a `mut self`, but a `&mut self`. Let me investigate a bit.

Reply:
Okay, it's not consumed by `compile`, but the mutable borrow lasts as long as `compiled_code`, which is causing the borrow error.